src/main/kotlin/org/koitharu/kotatsu/parsers/site/HoneyMangaParser.kt

```diff
@@ -0,0 +1,208 @@
+import org.koitharu.kotatsu.parsers.util.urlEncoded
+import java.text.SimpleDateFormat
+import java.util.EnumSet
+import java.util.Locale
+
+
+private const val PAGE_SIZE = 20
+
+@MangaSourceParser("HONEYMANGA", "Honey Manga", "uk")
+class HoneyMangaParser(context: MangaLoaderContext) : PagedMangaParser(context, MangaSource.HONEYMANGA, PAGE_SIZE), Interceptor {
+
+	private val urlApi get() = "https://data.api.$domain"
+	private val mangaApi get() = "$urlApi/v2/manga/cursor-list"
+	private val chapterApi get() = "$urlApi/v2/chapter/cursor-list"
+	private val genresListApi get() = "$urlApi/genres-tags/genres-list"
+	private val framesApi get() = "$urlApi/chapter/frames"
+	private val searchApi = "https://search.api.$domain/api/v1/title/search-matching?query="
+
+	private val imageStorageUrl = "https://manga-storage.fra1.digitaloceanspaces.com/public-resources"
+
+	override val headers
+		get() = Headers.Builder()
+			.add("User-Agent", "Mozilla/5.0 (Android 13; Mobile; rv:68.0) Gecko/68.0 Firefox/109.0")
+			.build()
+
+	override val configKeyDomain: ConfigKey.Domain
+		get() = ConfigKey.Domain("honey-manga.com.ua", null)
+
+	override val sortOrders: Set<SortOrder> = EnumSet.of(
+		SortOrder.POPULARITY,
+		SortOrder.NEWEST,
+	)
+
+	override suspend fun getDetails(manga: Manga): Manga {
+		val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.US)
+		val body = JSONObject()
+		body.put("mangaId", manga.url)
+		body.put("pageSize", 999999) // Hack lol (no)
+		body.put("page", 1)
+		body.put("sortOrder", "ASC")
+		val chapterRequest = webClient.httpPost(chapterApi, body).parseJson()
+		return manga.copy(
+			chapters = chapterRequest.getJSONArray("data").mapJSONIndexed() { i, jo ->
+				MangaChapter(
+					id = generateUid(jo.getString("id")),
+					name = buildString {
+						append("Том ")
+						append(jo.optString("volume", "0"))
+						append(". ")
+						append("Розділ ")
+						append(jo.optString("chapterNum", "0"))
+						if (jo.optString("title") != "Title") {
+							append(" - ")
+							append(jo.optString("title"))
+						}
+					},
+					number = i + 1,
+					url = jo.optString("chapterResourcesId"),
+					scanlator = null,
+					uploadDate = dateFormat.tryParse(jo.getString("lastUpdated")),
+					branch = null,
+					source = source
+				)
+			}
+		)
+	}
+
+	override suspend fun getListPage(
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder
+	): List<Manga> {
+		val body = JSONObject()
+		var content: JSONArray = JSONArray()
+
+		// Popular/Newest
+		content = if (query.isNullOrEmpty()) {
+			body.put("page", page)
+			body.put("pageSize", PAGE_SIZE)
+			body.put("filters", JSONArray())
+			val sort = JSONObject()
+			sort.put("sortBy", getSortKey(sortOrder))
+			sort.put("sortOrder", "DESC")
+			body.put("sort", sort)
+			webClient.httpPost(mangaApi, body).parseJson().getJSONArray("data")
+		// Search
+		} else {
+			if((query.length < 3) || (page > 1)) return emptyList()
```

Comment:
It might be better to throw `IllegalArgumentException` if query.length < 3 with a simple English message